batch/batch/batch.py

```diff
@@ -178,30 +184,35 @@ def job_record_to_dict(record, running_status=None):
 
     batch_id = record['batch_id']
     job_id = record['job_id']
+    attempt_id = record['attempt_id']
     id = (batch_id, job_id)
 
     instance_name = record['instance_name']
     assert instance_name is not None
 
-    log.info(f'unscheduling job {id} from instance {instance_name}')
-
-    instance = inst_pool.name_instance.get(instance_name)
-    if not instance:
-        log.warning(f'unschedule job {id}: unknown instance {instance_name}')
-        return
+    log.info(f'unscheduling job {id}, attempt {attempt_id} from instance {instance_name}')
 
     end_time = time_msecs()
-    await check_call_procedure(
-        db,
-        'CALL unschedule_job(%s, %s, %s, %s, %s);',
-        (batch_id, job_id, instance_name, end_time, 'cancelled'))
 
-    log.info(f'unschedule job {id}: updated database')
+    try:
+        rv = await db.execute_and_fetchone(
+            'CALL unschedule_job(%s, %s, %s, %s, %s, %s);',
+            (batch_id, job_id, attempt_id, instance_name, end_time, 'cancelled'))
+    except:
+        log.exception(f'error while unscheduling job {id} on instance {instance_name}')
+        raise
```

Comment:
an exception here would interrupt the ability of cancel_1 to cancel other jobs. I'm not sure exactly what kinds of errors can happen here? I guess if the DB is dieing, there's no point trying other things. But if unschedule_job has a bug or the database is in a weird state for one particular record, this could prevent any job from getting cancelled.

Feels like maybe we want a separate system that tracks DB-releated error statistics (and maybe a log of all the errors?) that we can query much like we can query the database's status with `SHOW ENGINE INNODB STATUS`. If  continuing isn't a correctness bug (like here, continuing to cancel other jobs is fine) then we don't error, we just note that the DB had an issue. We (hail team) should monitor this to check for database issues, but generally it's fine if the db occasionally has a hiccup.

Reply:
I think cancel needs to be fixed in a separate project to do things in parallel and be robust to individual unschedule jobs raising an exception.

Reply:
I added a try, except in the cancel loop